packages/@aws-cdk/aws-synthetics/lib/canary.ts

```diff
@@ -306,6 +314,7 @@ export class Canary extends cdk.Resource {
       failureRetentionPeriod: props.failureRetentionPeriod?.toDays(),
       successRetentionPeriod: props.successRetentionPeriod?.toDays(),
       code: this.createCode(props),
+      runConfig: this.createRunConfig(props),
```

Comment:
```suggestion
      runConfig: props.environmentVariables ?? {
              environmentVariables: props.environmentVariables,
       }
```

Reply:
are you suggesting that I ditch the `createRunConfig` function? or that the run config shouldn't exist if env vars aren't supplied (as this is already the case)